apps/els_core/src/els_protocol.erl

```diff
@@ -79,6 +80,12 @@ range(#{ from := {FromL, FromC}, to := {ToL, ToC} }) ->
    , 'end' => #{line => ToL - 1,   character => ToC - 1}
    }.
 
+-spec range_line(poi_range()) -> range().
```

Comment:
Really nice to handle this in a separate function. The idea behind the `els_protocol` module is to handle type conversions between the internal representation and the LSP one (it only support ranges right now), so a better place to implement this could be the `els_range`, which contains manipulation of ranges. So maybe have a function in `els_range` which selects the line and combine it with the `els_protocol:range/1` function? Something like:

```
LineRange = els_range:line(Range),
els_protocol:range(LineRange).
```

Reply:
address in c3849b7